pkg/api/handlers/compat/distribution.go

```diff
@@ -0,0 +1,88 @@
+//go:build !remote
+
+package compat
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/containers/image/v5/docker"
+	"github.com/containers/image/v5/docker/reference"
+	"github.com/containers/image/v5/manifest"
+	"github.com/containers/podman/v5/pkg/api/handlers/utils"
+	registrytypes "github.com/docker/docker/api/types/registry"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func InspectDistribution(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	name := utils.GetName(r)
+
+	ref, err := reference.ParseAnyReference(name)
+	if err != nil {
+		utils.Error(w, http.StatusBadRequest, fmt.Errorf("error parsing image reference %q: %w", name, err))
+		return
+	}
+
+	namedRef, ok := ref.(reference.Named)
+	if !ok {
+		if _, ok := ref.(reference.Digested); ok {
+			// full image ID
+			utils.Error(w, http.StatusBadRequest, fmt.Errorf("no manifest found for full image ID"))
+			return
+		}
+		utils.Error(w, http.StatusBadRequest, fmt.Errorf("unknown image reference format: %s", name))
+		return
+	}
+
+	namedRef = reference.TagNameOnly(namedRef)
+
+	imgRef, err := docker.NewReference(namedRef)
+	if err != nil {
+		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("error creating image reference: %w", err))
+		return
+	}
+
+	imgSrc, err := imgRef.NewImageSource(r.Context(), nil)
+	if err != nil {
+		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("error getting image source: %w", err))
+		return
+	}
+	defer imgSrc.Close()
+
+	manBlob, manType, err := imgSrc.GetManifest(r.Context(), nil)
+	if err != nil {
+		utils.Error(w, http.StatusInternalServerError, fmt.Errorf("error getting manifest: %w", err))
+		return
+	}
```

Comment:
not sure if NewImageSource() alone does some remote reeust alreayd but here we need to handle auto failures. the api requires us to return 401, i.e. see DockerClient.ErrUnauthorizedForCredentials in pkg/api/handlers/compat/auth.go

Reply:
Yes, `NewImageSource` looks for a reachable mirror.